velox/exec/HashTable.h

```diff
@@ -967,6 +980,11 @@ class HashTable : public BaseHashTable {
   // the table. This is set to 'capacity_ * sizeof(void*) - 1'.
   int64_t sizeMask_{0};
 
+  // The start bit position of the spill partition bits. This reflects the
+  // current processing batch's spill level's start bit, and could be changed
+  // when a different batch with a different spill level is processed.
+  int8_t spillInputStartPartitionBit_{kNoSpillInputStartPartitionBit};
```

Comment:
Ideally we should not keep a transient value like this as a field member.  How much work would it be to pass it as argument down the call chain?  If we only need to change private methods I would say let's just do that.

Reply:
I think current spill state can actually be part of the hashtable state instead of transient. Passing the argument down the call chain will make many member functions have an additional spill parameter that only makes sense at some call sites.

Reply:
Adding it as class member could create potential bug that we would forget to update it somewhere.  Argument would be the best in this case because the lifecycle of the value fits nicely during one call.  For those calls happening outside spill context, `kNoSpillInputStartPartitionBit` should be used.

Reply:
That is correct. In fact imagine no spill is also a spill state if not set. The spill bits flag is only used for overlapping checking so whenever the spill bit is kNoSpill we don't do the check (which we can add). There 3 or 4 public APIs that are the entrance of setting spill state and that should be not too easy to miss. 

If we decide to propagate parameters then following methods will need to change

setHashMode()
checkSize()
allocateTables()
rehash()

and maybe more. For most of the methods, having that spill bits parameter does not really make sense and will make those private methods look confusing.

Reply:
These are all private methods, I would say just add it as parameter.  More state is more potential bug.